reconcile/utils/saasherder.py

```diff
@@ -174,7 +174,7 @@ def __init__(
         integration_version,
         settings,
         jenkins_map=None,
-        accounts=None,
+        initialise_state=False,
```

Comment:
I would prefer a `state: Optional[State]` instead of `initialise_state: bool`

Reply:
me too. this PR is minimal on purpose so things like injecting`SecretReader` and even the`State` itself will happen in the followup MR if that is ok with you